### PR TITLE
Move to netstandard2.0

### PR DESCRIPTION
**Pull request by jaredpar**
This moves the project to use netstandard2.0 as the base target
framework for projects. There is no value in maintaining netstandard1.1
and net20 anymore.

### CODE REVIEW

File: src/Microsoft.DiaSymReader/SymUnmanagedFactory.cs [hunk 1]
```diff
@@ -79,18 +78,13 @@ internal static string DiaSymReaderModuleName
...
         });
-#endif
 
         // internal for testing
         internal static string GetEnvironmentVariable(string name)
         {
             try
             {
-#if NET20
-                return Environment.GetEnvironmentVariable(name);
```
Comment (tmat) [inline src/Microsoft.DiaSymReader/SymUnmanagedFactory.cs:90]: >Environment.GetEnvironmentVariable(name); [](http://example.com/codeflow?start=23&length=41)

This was a workaround for netstandard1.1, the NET20 code is certainly better :)

**Comment by tmat** [comment]
Let's also set version here: https://github.com/dotnet/symreader/blob/main/eng/Versions.props#L6 tp 2.0.0 since this is a breaking change.

File: src/Microsoft.DiaSymReader/Utilities/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,12 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the License.txt file in the project root for more information.
-#if NET20
-
-[assembly: System.Runtime.Versioning.TargetFramework(".NETFramework,Version=v2.0")]
-
-#else
-
-[assembly: System.Security.AllowPartiallyTrustedCallers]
```
Comment (jaredpar) [inline src/Microsoft.DiaSymReader/Utilities/AssemblyInfo.cs:10]: @tmat want to make sure u see that I deleted this. It seems to serve no purpose and just cause issues. Want to make sure I'm not missing a reason for it being there.

Comment (tmat) [reply src/Microsoft.DiaSymReader/Utilities/AssemblyInfo.cs:10]: I don't recall why we had it. Seems unnecessary for netstandard2.0.

**Comment by jaredpar** [comment]
@tmat version updated

File: src/Microsoft.DiaSymReader/SymUnmanagedFactory.cs (hunk 1 above)
Comment (jaredpar) [reply src/Microsoft.DiaSymReader/SymUnmanagedFactory.cs:90]: Ah okay. Will fix that.

**Review by tmat — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Sorry for commenting on an old PR but any idea why this change never shipped to customers? NuGet's latest version is still 1.4.0.

**Comment by jaredpar** [comment]
@tmat will need to answer as I'm unfamiliar with how NuPkg are produced from this repo

**Comment by ViktorHofer** [comment]
@tmat would you know the answer to my above question?

**Comment by tmat** [comment]
@jaredpar This repo changes very rarely. We don't produce release versions regularly.

**Comment by ViktorHofer** [comment]
Can we please prioritize releasing a new version with the netstandard2.0 change? The existing package version brings in NETStandard.Library/1.6.1 which itself transitively brings in the entire .NET Standard 1.6 dependency graph (which is huge). That graph might contains either deprecated or vulnerable packages which we haven't touched for years.

I noticed this as I looked into a package that depends on Microsoft.DiaSymReader and the reference caused Component Governance warnings.

**Pull request merged**